### ISSUE TITLE
No JavaScript output

### ISSUE THREAD

**Issue by tangentlin** [ivogabe/gulp-typescript#126]
I have got the 2.7.4 of gulp-typescript and all of a sudden I don't see any JavaScript output or any errors that might help me troubleshoot.

Below are relevant sources:

### gulpfile.js

```javascript
var gulp = require('gulp');
var ts = require('gulp-typescript');

var sourcemaps = require('gulp-sourcemaps');

var source = {
    path: "."
};

var filePattern = {
    all: "/**/*.*",
    html: "/**/*.html",
    typescript: "/**/*.ts",
    excludeExternalDependencies: ["!bower_components/**/*", "!node_modules/**/*"]
};

var webserverConfig = {
    root: '../out',
    host: 'localhost',
    port: 8000,
    defaultPage: 'index.html'
};

var tsProject = ts.createProject('./tsconfig.json');

gulp.task("script", function() {

    var tsResult = gulp.src([source.path + filePattern.typescript].concat(filePattern.excludeExternalDependencies))
        .pipe(ts(tsProject));

    return tsResult.js.pipe(gulp.dest(webserverConfig.root));

});
```

### tsconfig.json

```json
{
  "compilerOptions": {
    "declarationFiles": false,
    "module": "amd",
    "noImplicitAny": false,
    "removeComments": false,
    "preserveConstEnums": true,
    "target": "es5",
    "out": "../out",
    "sourceMap": true
  }
}
```

**Comment by ivogabe** [comment]
I don't get your configuration... Why are you using external modules in combination with the `out` option? I think you need `outDir` instead of `out`. Using `"outDir": "../out"` or `"out": "out.js"` (and removing `"module": "amd"`) will solve it. You can't use `out` and `module` together.

**Comment by tangentlin** [comment]
Thanks, changing ```out``` to ```outDir``` helps inch further, it displays an error which puzzles me

> The common source directory of the source files isn't equal to the common base directory of the input files. That isn't supported using the version of TypeScript currently used. Use a newer version of TypeScript instead or have the common source directory point to the common base directory.

How are _common source directory_ and _common base directory_ determined?  Currently I have the project setup as

```
--scripts  (all the project ts files)
 |-typings (d.ts from tsd) 
 |-bower_components (external js files)
 |-node_modules (external js files)
 |-tsconfig.json
 |-gulpfile.js
```

**Comment by ivogabe** [comment]
"common" means the longest path that all files/directories share, eg `foo/bar/baz` and `foo/bar/foo` will be turned into `foo/bar`. The common base directory is the common directory of all base paths of the input source files (only .ts, not .d.ts). The base path is usually where the glob starts in `gulp.src`. In this case that's `./`. The common source directory is calculated by TypeScript and refers to the common directory of all source files, including resolved files. In your case, the source files look to be stored in a directory under the root. You have to make that the base path, like this:
```javascript
gulp.src([
    'src/**/*.ts', // common base path will be 'src'
    'typings/**/*.d.ts'
])
```
Or you have to use TypeScript 1.5, since this only applies to TS1.4.

**Comment by ivogabe** [comment]
I've added a fix such that the common source directory will always be calculated properly. The fix is released in version 2.7.5. I'm closing this now as your project looks to be compiling if you use v2.7.5. If it didn't work for you, I'll reopen this.
